encoders/numeric/CompressionVaeEncoder/__init__.py

```diff
@@ -33,42 +34,45 @@ def post_init(self):
         import cvae.lib.model_iaf as model
         import tensorflow as tf
 
-        params_path = os.path.join(self.model_path, 'params.json')
-
-        config = tf.ConfigProto(log_device_placement=False)
-        config.gpu_options.allow_growth = True
-
-        with tf.Graph().as_default():
-            if os.path.exists(params_path):
-                # Load parameter file.
-                with open(params_path, 'r') as f:
-                    param = json.load(f)
-
-                net = model.VAEModel(param,
-                                     None,
-                                     input_dim=param['dim_feature'],
-                                     keep_prob=tf.placeholder_with_default(input=tf.cast(1.0, dtype=tf.float32),
-                                                                           shape=(),
-                                                                           name="KeepProb"),
-                                     initializer='orthogonal')
-                # Placeholder for data features
-                self.data_feature_placeholder = tf.placeholder_with_default(
-                    input=tf.zeros([64, param['dim_feature']], dtype=tf.float32),
-                    shape=[None, param['dim_feature']])
-
-                self.embeddings = net.embed(self.data_feature_placeholder)
-
-                self.sess = tf.Session(config=config)
-                init = tf.global_variables_initializer()
-                self.sess.run(init)
-
-                # Saver for loading checkpoints of the model.
-                saver = tf.train.Saver(var_list=tf.trainable_variables())
-                cvae.load(saver, self.sess, self.model_path)
-
-                self.to_device()
-            else:
-                raise FileNotFoundError(f'The {params_path} doesn\'t exist')
+        if self.model_path and os.path.exists(self.model_path):
```

Comment:
It seems that the new logic and the FileNotFound exception should have the same logic, is there a way to unify it? Maybe checking model path and params path before session run?